### PR TITLE
Ensure constant-time fake credential response

### LINKED ISSUE #219: Give constant-time trick in current user enumeration mitigation scheme
To ensure a constant time "fake" response by the server, two solutions are suggested

- the server decides on a seed at setup, and deterministically generates a public key at each fake response to use in a fake client record. The server remembers the seed.
- the server generates a public key that it will use in each fake response. The server remembers that public key.

### PR DESCRIPTION
**Pull request by bytemare**
Closes #219

### CODE REVIEW

File: draft-irtf-cfrg-opaque.md [hunk 1]
```diff
@@ -1155,14 +1155,22 @@ Steps:
 8. Output response
 ~~~
 
-In the case of a record that does not exist, the server SHOULD invoke the
-CreateCredentialResponse function where the record argument is configured so that:
+In the case of a record that does not exist and if client enumeration prevention is desired,
+the server must respond to the credential request to fake the existence of the record.
```
Comment (chris-wood) [inline draft-irtf-cfrg-opaque.md:1159]:
```suggestion
the server MUST respond to the credential request to fake the existence of the record.
```

File: draft-irtf-cfrg-opaque.md [hunk 2]
```diff
@@ -1155,14 +1155,22 @@ Steps:
...
 - record.masking_key is set to a random byte string of length Nh
 - record.envelope is set to the byte string consisting only of zeros of length Ne
 
-In the generated test vectors from {{fake-vectors}}, we use `client_public_key` for the record's public key, and `masking_key` for the record's masking key parameter.
+It is RECOMMENDED that a fake client record is created once (e.g. as the first user record
+of the application), and then readily available during the application's lifetime. This ensures
+relative constant-time access and processing of the record compared to a legitimate client record,
+avoiding timing leaks.
+
+In the generated test vectors from {{fake-vectors}}, we use `client_public_key` for this record's
```
Comment (chris-wood) [inline draft-irtf-cfrg-opaque.md:1172]: This should move to the test vector section, rather than be listed here.

File: draft-irtf-cfrg-opaque.md [hunk 3]
```diff
@@ -1155,14 +1155,22 @@ Steps:
...
 
 - record.client_public_key is set to a randomly generated public key of length Npk
 - record.masking_key is set to a random byte string of length Nh
 - record.envelope is set to the byte string consisting only of zeros of length Ne
 
-In the generated test vectors from {{fake-vectors}}, we use `client_public_key` for the record's public key, and `masking_key` for the record's masking key parameter.
+It is RECOMMENDED that a fake client record is created once (e.g. as the first user record
+of the application), and then readily available during the application's lifetime. This ensures
+relative constant-time access and processing of the record compared to a legitimate client record,
+avoiding timing leaks.
```
Comment (chris-wood) [inline draft-irtf-cfrg-opaque.md:1170]:
```suggestion
It is RECOMMENDED that a fake client record is created once (e.g. as the first user record
of the application) and stored alongside legitimate client records. This allows servers to locate
the record in time comparable to that of a legitimate client record.
```

**Review by chris-wood — CHANGES_REQUESTED** [review]

**Comment by chris-wood** [comment]
@bytemare, sorry, I meant to suggest that the comment could go [here](https://cfrg.github.io/draft-irtf-cfrg-opaque/draft-irtf-cfrg-opaque.html#section-appendix.c), at the end of the text in the start of Appendix C.

**Review by chris-wood — CHANGES_REQUESTED** [review]
Can we move the text up to the start of Appendix C?

**Comment by bytemare** [comment]
> @bytemare, sorry, I meant to suggest that the comment could go [here](https://cfrg.github.io/draft-irtf-cfrg-opaque/draft-irtf-cfrg-opaque.html#section-appendix.c), at the end of the text in the start of Appendix C.

Ah, of course, you mean "Test Vectors", right?

**Review by chris-wood — APPROVED** [review]

**Pull request merged**